app/src/main/java/de/westnordost/streetcomplete/view/controller/StreetSideSelectWithLastAnswerButtonViewController.kt

```diff
@@ -171,38 +175,59 @@ class StreetSideSelectWithLastAnswerButtonViewController<I>(
+                    putString("$lastSelectionPreferencePrefix.right", r?.let { serializeLastSelection(it, true) })
+                }
+                Sides.LEFT -> putString("$lastSelectionPreferencePrefix.only_left", l?.let { serializeLastSelection(it, false) })
+                Sides.RIGHT -> putString("$lastSelectionPreferencePrefix.only_right", r?.let { serializeLastSelection(it, true) })
+            }
         }
     }
 
     private fun updateLastSelectionButton() {
         updateLastSelectionButtonVisibility()
-        if (showSides != Sides.RIGHT) {
-            lastSelectionLeft?.let { lastAnswerButtonBinding.leftSideImageView.setImage(it.icon) }
-        }
-        if (showSides != Sides.LEFT) {
-            lastSelectionRight?.let { lastAnswerButtonBinding.rightSideImageView.setImage(it.icon) }
+        when (showSides) {
+            Sides.BOTH -> {
+                lastSelectionLeft?.let { lastAnswerButtonBinding.leftSideImageView.setImage(it.icon) }
+                lastSelectionRight?.let { lastAnswerButtonBinding.rightSideImageView.setImage(it.icon) }
+            }
+            Sides.LEFT -> lastSelectionOnlyLeft?.let {
+                lastAnswerButtonBinding.leftSideImageView.setImage(it.icon)
+                lastAnswerButtonBinding.rightSideImageView.setImage(null)
+            }
+            Sides.RIGHT ->
+                lastSelectionOnlyRight?.let {
+                    lastAnswerButtonBinding.rightSideImageView.setImage(it.icon)
+                    lastAnswerButtonBinding.leftSideImageView.setImage(null)
+                }
         }
     }
 
     private fun updateLastSelectionButtonVisibility() {
         lastAnswerButtonBinding.root.isGone =
             when (showSides) {
                 Sides.BOTH -> lastSelectionLeft == null && lastSelectionRight == null
-                Sides.LEFT -> lastSelectionLeft == null
-                Sides.RIGHT -> lastSelectionRight == null
+                Sides.LEFT -> lastSelectionOnlyLeft == null
+                Sides.RIGHT -> lastSelectionOnlyRight == null
             } ||
             left != null ||
             right != null ||
-            showSides != Sides.BOTH ||
             !puzzleView.isEnabled
     }
 
     private fun applyLastSelection() {
         val isUpsideDown = isStreetDisplayedUpsideDown()
-        val l = if (isUpsideDown) lastSelectionRight else lastSelectionLeft
-        val r = if (isUpsideDown) lastSelectionLeft else lastSelectionRight
+        val l: StreetSideDisplayItem<I>?
+        val r: StreetSideDisplayItem<I>?
+        if (showSides == Sides.BOTH) {
+            l = if (isUpsideDown) lastSelectionRight else lastSelectionLeft
+            r = if (isUpsideDown) lastSelectionLeft else lastSelectionRight
+        } else {
+            l = if (isUpsideDown) lastSelectionOnlyRight else lastSelectionOnlyLeft
+            r = if (isUpsideDown) lastSelectionOnlyLeft else lastSelectionOnlyRight
```

Comment:
But, does that work? Let's say you have 
- lastSelectionOnlyLeft  = cycle track
- lastSelectionOnlyRight = cycle lane


Now, there is a oneway road for which only the right side is displayed. Hence, the last answer button is displayed as cycle lane. But then, you rotate the road upside down. Now, according to this logic, tapping on the last answer button which **displays** a cycle lane is going to add a cycle track, no?

Reply:
Meanwhile I also found some cases where this didn't work properly... I'll have a look at it later (converting to draft while this is not solved)

Reply:
I think we did establish in the main discussion that the left/right serialization is a relict from the past that is actually not used anymore anywhere. So maybe the better solution would be to have just one `lastSelectionOneSideOnly` instead.

Reply:
I removed the serialization, and now only one value is stored. But I didn't switch to `lastSelectionOneSideOnly`, because in deserialization the side is relevant.

Reply:
Oh.

So that's why there is this `isRight: Boolean` in both the serializer and deserializer

Reply:
How do you changes solve the issue that when you turn the map (at most) 180°, a different not previously shown item is suddenly depicted on the button?

Reply:
The shown last solution for one side is the same (from same settings value), only deserialized for the correct side.

Reply:
I don't understand.

Reply:
```kotlin
        lastSelectionOnlyLeft = prefs.getString("$lastSelectionPreferencePrefix.oneSide", null)?.let { deserializeLastSelection(it, false) }
        lastSelectionOnlyRight = prefs.getString("$lastSelectionPreferencePrefix.oneSide", null)?.let { deserializeLastSelection(it, true) }
```
Both `lastSelectionOnlyLeft` and `lastSelectionOnlyRight` use the value stored in `"$lastSelectionPreferencePrefix.oneSide"`. The only difference is whether `isRight` is set to true in `deserializeLastSelection`